src/TesApi.Web/Runner/NodeTaskBuilder.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Tes.Runner.Models;
+
+namespace TesApi.Web.Runner
+{
+    /// <summary>
+    /// Builder of NodeTask
+    /// </summary>
+    public class NodeTaskBuilder
+    {
+        private readonly NodeTask nodeTask;
+        const string NodeTaskOutputsMetricsFormat = "FileUploadSizeInBytes={Size}";
+        const string NodeTaskInputsMetricsFormat = "FileDownloadSizeInBytes={Size}";
+
+        /// <summary>
+        /// Creates a new builder
+        /// </summary>
+        public NodeTaskBuilder()
+        {
+            nodeTask = new NodeTask();
+
+            SetDefaultValues();
+        }
+
+        /// <summary>
+        /// Creates a new builder with an existing NodeTask
+        /// </summary>
+        /// <param name="nodeTask"></param>
+        public NodeTaskBuilder(NodeTask nodeTask)
+        {
+            this.nodeTask = nodeTask;
+        }
+
+        private void SetDefaultValues()
+        {
+            nodeTask.ContainerWorkDir = "/";
```

Comment:
If the container's workdir is not explicitly provided in the TesTask, we don't set it in the docker command-line (or API) (we used to set it to "/", but that was changed over a month ago).